### PR TITLE
feat: add Jenkinsfile for comparision job

### PR DESCRIPTION
**Pull request by zdtsw**
Add new Jenkinsfile for a new job to do simple comparision of two downloadable tar ball

Ref https://github.com/adoptium/ci-jenkins-pipelines/issues/301

### CODE REVIEW

**Review by karianna — COMMENTED** [review]
The comparison tool is diff - does that go deep enough?

File: tools/reproduce_comparison/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+Jenkins job does comparision for two different build artifacts
+If they are identical, build pass; or fail
+*/
+pipeline {
+    agent {
+        label linux
+    }
+    parameters {
+        string(name: 'URL1', defaultValue: '', description: 'URL to one versoin of the build artifacts\ne.g: https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk18-pipeline/lastSuccessfulBuild/artifact/target/linux/x64/temurin/OpenJDK18U-jdk_x64_linux_hotspot_2022-06-11-23-30.tar.gz')
```
Comment (karianna) [inline tools/reproduce_comparison/Jenkinsfile:10]:
```suggestion
        string(name: 'URL1', defaultValue: '', description: 'URL to one version of the build artifacts\ne.g.: https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk18-pipeline/lastSuccessfulBuild/artifact/target/linux/x64/temurin/OpenJDK18U-jdk_x64_linux_hotspot_2022-06-11-23-30.tar.gz')
```

File: tools/reproduce_comparison/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        string(name: 'URL2', defaultValue: '', description: 'URL to the other versoin of the build artifacts\ne.g: https://github.com/adoptium/temurin19-binaries/releases/download/jdk-2022-06-09-19-11-beta/OpenJDK-jdk_x64_linux_hotspot_2022-06-09-03-31.tar.gz')
```
Comment (karianna) [inline tools/reproduce_comparison/Jenkinsfile:11]:
```suggestion
        string(name: 'URL2', defaultValue: '', description: 'URL to the other version of the build artifacts\ne.g.: https://github.com/adoptium/temurin19-binaries/releases/download/jdk-2022-06-09-19-11-beta/OpenJDK-jdk_x64_linux_hotspot_2022-06-09-03-31.tar.gz')
```

**Review by karianna — CHANGES_REQUESTED** [review]

**Comment by zdtsw** [comment]
> The comparison tool is diff - does that go deep enough?
diff 
```
-r, --recursive
              recursively compare any subdirectories found
```

````
$ diff -q -r wen1 wen2
Files wen1/filewen11 and wen2/filewen11 differ
Only in wen2/folderwen11: filefolder11
Only in wen2/folderwen11: folderwen12



$ tree wen1
wen1
├── filewen11
├── filewen12
└── folderwen11
    ├── filefolderwen11
    └── filefolderwen12

1 directory, 4 files



$ tree wen2
wen2
├── filewen11
├── filewen12
└── folderwen11
    ├── filefolder11
    ├── filefolderwen11
    ├── filefolderwen12
    └── folderwen12

2 directories, 5 files
```

**Review by sxa — COMMENTED** [review]
This seems to do a `cleanWs()` before the comparison but is it going to leave the artifacts around once finished? Sine this can execute on any machine labelled linux it's going to leave lots of stuff around on all of those machines I believe, which is undesirable.

**Comment by zdtsw** [comment]
> This seems to do a `cleanWs()` before the comparison but is it going to leave the artifacts around once finished? Sine this can execute on any machine labelled linux it's going to leave lots of stuff around on all of those machines I believe, which is undesirable.

can add another cleanws() then the job is done, maybe better to cleanup space?

**Review by andrew-m-leonard — APPROVED** [review]

**Pull request merged**